src/main/java/com/github/scr/j8iterables/core/CloseableSpliterator.java

```diff
@@ -0,0 +1,72 @@
+package com.github.scr.j8iterables.core;
+
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.util.Comparator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+/**
+ * A {@link Spliterator} that closes a {@link Closeable} resource when it is completed.
```

Comment:
Here the doc is kinda of misleading, because the code says resource is can be any type other than `Closable`.